### ISSUE TITLE
Cannot read property 'then' of undefined

### ISSUE THREAD

**Issue by smenshikov** [effector/effector#511]
**What is the current behavior:**
Get an error "Cannot read property 'then' of undefined"

My app based on Next.js. There is a component called `SimilarProductsList` that uses `useEvent` hook like this
  `const boundGetSimilarProductsFx = useEvent(getSimilarProductsFx);`

`getSimilarProductsFx` is a simple effect that wraps an API endpoint.

Also, the `SimilarProductsList` has a `useEffect` hook

```
useEffect(() => {
   boundGetSimilarProductsFx(productId)
        .then((products) => {
           setProducts(products);
        })
  }, []);
```

`setProducts` is a function to update the local state of the component like this

  `const [products, setProducts] = useState<ISimilarProduct[]>([]);`

So, when I run the project the `useEffect` hook starts on the browser side and I see this

![image](https://user-images.githubusercontent.com/75778692/129940949-37d9ad33-3933-42a5-9456-08c626e7e91d.png)

**Please provide the steps to reproduce and if possible a minimal demo of the problem via https://share.effector.dev, https://codesandbox.io or similar**

**What is the expected behavior:**
Should work fine as previously. 

**Which versions of effector packages, and which browser and OS are affected by this issue? Did this work in previous versions of effector?:**

the Effector version 21.8.11 and effector-react 21.3.2 work fine without the error. Today I switched on this version https://github.com/effector/effector/packages/21591 effector 21.8.12-4d855200 and https://github.com/effector/effector/packages/43968 21.3.3-4d855200 and the error appeared.

**Comment by zerobias** [comment]
Clearly this is a bug, will fix than in next effector/effector-react release

**Comment by zerobias** [comment]
Fixed in canary build [bcc8574e](https://github.com/effector/effector/packages/21591?version=21.8.12-bcc8574e)
